test/integration_test.js

```diff
@@ -0,0 +1,95 @@
+};
+let clientService;
+
+describe('Integration test on ClientService', async () => {
+  const mockedFunctionId = 'TestFunction';
+  const mockedContractId = `StateUpdater${Date.now()}`;
+  const mockedContractName = 'com.org1.contract.StateUpdater';
+  const mockedFunctionName = 'com.org1.function.TestFunction';
+  const mockedAssetId = 'mockedAssetId';
+  const mockedState = 1;
+  const mockedContractArgument = {
+    asset_id: mockedAssetId,
+    state: mockedState,
+  };
+  const mockedFunctionArgument = {
+    asset_id: mockedAssetId, // passed via contractArgument
+    state: mockedState,
+    _function_: mockedFunctionId,
+  };
+  const property = {
+    asset_id: mockedAssetId,
+    state: mockedState,
+  };
+  const mockedByteContract = fs.readFileSync(
+      __dirname + '/StateUpdater.class');
+  const mockedByteFunction = fs.readFileSync(
+      __dirname + '/TestFunction.class');
+  clientService = new ClientService(properties);
+  describe('registerCertificate', () => {
+    it('should works as expected', async () => {
+      const response = await clientService.registerCertificate();
+      assert.equal(response.getStatus(), 200);
+    });
+  });
+  describe('registerFunction', () => {
+    it('should works as expected', async () => {
+      const response = await clientService.registerFunction(mockedFunctionId,
+          mockedFunctionName,
+          mockedByteFunction);
+      assert.equal(response.getStatus(), 200);
+    });
+  });
+  describe('registerContract', () => {
+    it('should works as expected', async () => {
+      const response = await clientService.registerContract(mockedContractId,
+          mockedContractName,
+          mockedByteContract, property);
+      assert.equal(response.getStatus(), 200);
+    });
+  });
+  describe('listContracts', () => {
+    it('should works as expected', async () => {
+      const response = await clientService.listContracts(mockedContractId);
+      assert.equal(response.getStatus(), 200);
+    });
+  });
+  describe('executeContract', () => {
+    it('should works as expected when execute a registered Contract',
+        async () => {
+          const response = await clientService.executeContract(mockedContractId,
```

Comment:
Eventually, the integration does not validate if the SDK is transmitting properly to the DLT server the contract parameters (passed when executing a contract) and the contract property (set when registering a contract).
To make sure this works, we could create a contract implementation that would just return back to the client in the result the "contract arguments" and "property". Then we would just have to assert the contract response contains the right "contract arguments" and "property".
What do you think ?

Reply:
Yup i agree, i will update the `StateUpdate.class` to return the provided argument.